universal_binary.go

```diff
@@ -48,6 +48,16 @@ type ExtractedFile struct {
 	UniversalArchInfo
 }
 
+type SectionReader interface {
+	io.Reader
+	io.ReaderAt
+}
+
+type ExtractedReader struct {
```

Comment:
super nit: we don't have to expose the unified interface if ultimately the purpose is to raise up one or more kinds of readers:
```golang

type ExtractedReader struct {
	Reader interface {
		io.Reader
		io.ReaderAt
	}
	UniversalArchHeader
}

```

then delete the `SectionReader` definition (which has a problematic name anyway)